builtin/logical/database/rotation_test.go

```diff
@@ -850,3 +997,122 @@ func newBoolPtr(b bool) *bool {
 	v := b
 	return &v
 }
+
+// copied from plugins/database/mongodb/mongodb_test.go
+// maybe move to a new builtin/logical/database/dbplugin/helper
+func prepareMongoDBTestContainer(t *testing.T) (cleanup func(), retURL string) {
+	if os.Getenv("MONGODB_URL") != "" {
+		return func() {}, os.Getenv("MONGODB_URL")
+	}
+
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		t.Fatalf("Failed to connect to docker: %s", err)
+	}
+
+	resource, err := pool.Run("mongo", "latest", []string{})
+	if err != nil {
+		t.Fatalf("Could not start local mongo docker container: %s", err)
+	}
+
+	cleanup = func() {
+		err := pool.Purge(resource)
+		if err != nil {
+			t.Fatalf("Failed to cleanup local container: %s", err)
+		}
+	}
+
+	retURL = fmt.Sprintf("mongodb://localhost:%s/vaulttestdb", resource.GetPort("27017/tcp"))
+
+	// exponential backoff-retry
+	if err = pool.Retry(func() error {
+		var err error
+		dialInfo, err := parseMongoURL(retURL)
+		if err != nil {
+			return err
+		}
+
+		session, err := mgo.DialWithInfo(dialInfo)
+		if err != nil {
+			return err
+		}
+		defer session.Close()
+		session.SetSyncTimeout(1 * time.Minute)
+		session.SetSocketTimeout(1 * time.Minute)
+		return session.Ping()
+	}); err != nil {
+		cleanup()
+		t.Fatalf("Could not connect to mongo docker container: %s", err)
+	}
+
+	return
+}
+
+// copied from plugins/database/mongodb/connection_producer.go
+func parseMongoURL(rawURL string) (*mgo.DialInfo, error) {
```

Comment:
Maybe we can add this to https://github.com/hashicorp/vault/blob/master/helper/testhelpers/mongodb/mongodbhelper.go

Reply:
Done in df5574a23 !